### PR TITLE
fix: sale status on sale screen

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **Bugfix**

### Description
Fix missing register to bid button on sale screen when the `registrationEndAt` is available.

**Before**
Screenshot 2020-11-10 at 14 14 47

**After**
Screenshot 2020-11-10 at 14 12 45

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

### CODE REVIEW

File: src/lib/Scenes/Sale/__tests__/helpers-tests.ts [hunk 1]
```diff
@@ -6,10 +6,11 @@ const dAdd = (date: number, amount: number, type: "day" | "days") => moment(date
...
 
 describe(saleStatus, () => {
-  it("calculates correctly", () => {
+  it("returns the right sale status", () => {
     const now = Date.now()
-    expect(saleStatus(dAdd(now, 1, "day"), dAdd(now, 2, "days"))).toBe("notYetOpen")
-    expect(saleStatus(dSub(now, 1, "day"), dAdd(now, 2, "days"))).toBe("active")
-    expect(saleStatus(dSub(now, 3, "days"), dSub(now, 2, "days"))).toBe("closed")
+    expect(saleStatus(dAdd(now, 1, "day"), dAdd(now, 2, "days"), dAdd(now, 1, "day"))).toBe("notYetOpen")
+    expect(saleStatus(dSub(now, 1, "day"), dAdd(now, 2, "days"), dAdd(now, 1, "day"))).toBe("active")
```
Comment (pvinis) [inline src/lib/Scenes/Sale/__tests__/helpers-tests.ts:12]: can we add an extra test for each of these with `null` in the third arg as well? just to have all our bases covered. 🙏

**Review by pvinis — APPROVED** [review]
I only would like two extra tests, then we can merge this 👍

File: src/lib/Scenes/Sale/__tests__/helpers-tests.ts (hunk 1 above)
Comment (MounirDhahri) [reply src/lib/Scenes/Sale/__tests__/helpers-tests.ts:12]: yes sure 🎉

File: src/lib/Scenes/Sale/__tests__/helpers-tests.ts [hunk 2]
```diff
@@ -6,10 +6,13 @@ const dAdd = (date: number, amount: number, type: "day" | "days") => moment(date
...
     const now = Date.now()
-    expect(saleStatus(dAdd(now, 1, "day"), dAdd(now, 2, "days"))).toBe("notYetOpen")
-    expect(saleStatus(dSub(now, 1, "day"), dAdd(now, 2, "days"))).toBe("active")
-    expect(saleStatus(dSub(now, 3, "days"), dSub(now, 2, "days"))).toBe("closed")
+    expect(saleStatus(dAdd(now, 1, "day"), dAdd(now, 2, "days"), dAdd(now, 1, "day"))).toBe("notYetOpen")
+    expect(saleStatus(dAdd(now, 1, "day"), dAdd(now, 2, "days"), null)).toBe("notYetOpen")
+    expect(saleStatus(dSub(now, 1, "day"), dAdd(now, 2, "days"), dAdd(now, 1, "day"))).toBe("active")
+    expect(saleStatus(dSub(now, 1, "day"), dAdd(now, 2, "days"), null)).toBe("active")
+    expect(saleStatus(dSub(now, 3, "days"), dSub(now, 2, "days"), null)).toBe("closed")
+    expect(saleStatus(dSub(now, 3, "days"), dAdd(now, 2, "days"), dSub(now, 1, "day"))).toBe("closed")
```
Comment (pvinis) [inline src/lib/Scenes/Sale/__tests__/helpers-tests.ts:16]:
```suggestion
    expect(saleStatus(dSub(now, 3, "days"), dAdd(now, 2, "days"), dSub(now, 1, "day"))).toBe("closed")
    expect(saleStatus(dSub(now, 3, "days"), dSub(now, 2, "days"), null)).toBe("closed")
```
Just for OCD's sake :D

**Comment by pvinis** [comment]
#squashongreen nice nice
